### ISSUE TITLE
Add PHAR distribution

### ISSUE THREAD

**Issue by llaville** [shipmonk-rnd/composer-dependency-analyser#169]
Hello,

Most PHP tools like this one make a phar version available for download.
It would be nice to see the same here.

Building a PHAR distribution is easy with https://github.com/box-project/box, and in case of your project, it requires no specific configuration.

For example :

## Compile a PHAR distribution

```text
Box version 4.6.2@29c3585 2024-04-23 19:35:41 UTC

 // Loading without a configuration file.

🔨  Building the PHAR "/shared/backups/forks/composer-dependency-analyser/bin/composer-dependency-analyser.phar"

? Checking Composer compatibility
    > Supported version detected
? No compactor to register
? Adding main file: /shared/backups/forks/composer-dependency-analyser/bin/composer-dependency-analyser
? Adding requirements checker
? Adding binary files
    > No file found
? Auto-discover files? Yes
? Exclude dev files? Yes
? Adding files
    > 31 file(s)
? Generating new stub
  - Using shebang line: #!/usr/bin/env php
  - Using banner:
    > Generated by Humbug Box 4.6.2@29c3585.
    >
    > @link https://github.com/humbug/box
? Dumping the Composer autoloader
? Removing the Composer dump artefacts
? No compression
? Setting file permissions to 0755
* Done.

No recommendation found.
No warning found.

 // PHAR: 73 files (313.56KB)
 // You can inspect the generated PHAR with the "info" command.

 // Memory usage: 15.47MB (peak: 15.58MB), time: <1sec
```

## Try it

```shell
devilbox@php-8.2.21 in /shared/backups/forks/composer-dependency-analyser $ bin/composer-dependency-analyser.phar
```

And you will get following output :

```text
Using config /shared/backups/forks/composer-dependency-analyser/composer-dependency-analyser.php

Detected multiple class loaders:
 • /shared/backups/forks/composer-dependency-analyser/vendor
 • phar:///shared/backups/forks/composer-dependency-analyser/bin/composer-dependency-analyser.phar/.box/vendor


No composer issues found
(scanned 39 files in 0.017 s)
```

## Learn more

And if you want to have a manifest in your phar version, I suggest to use my application https://github.com/llaville/box-manifest that is compatible with official BOX v4.

No specific configuration is also supported if you have a `box.json` config file like this one : 

```json
{
}
```

An to add, at least a console Manifest format, add this `box.json.dist` config file to your project : 

```json
{
    "stub": "stub.php",
    "files-bin": ["console.txt"],
    "map": [
        { "console.txt": ".box.manifests/console.txt" }
    ]
}
```

You need to download my application 

```shell
curl -Ls https://github.com/llaville/box-manifest/releases/latest/download/box-manifest.phar -o /usr/local/bin/box-manifest
chmod +x /usr/local/bin/box-manifest
```

And make the manifest content `console.txt` with a little bootstrapping php code `stub.php`

```shell
box-manifest manifest:build --ansi -vv -c box.json --output-file=console.txt --format console
box-manifest manifest:stub  --ansi -vv -c box.json --output-file=stub.php --resource console.txt
```

Then compile, as usual, with default BOX project 

```shell
box compile -c box.json.dist
```
to get 

```text
Box version 4.6.2@29c3585 2024-04-23 19:35:41 UTC

 // Loading the configuration file "box.json.dist".

🔨  Building the PHAR "/shared/backups/forks/composer-dependency-analyser/bin/composer-dependency-analyser.phar"

? Removing the existing PHAR "/shared/backups/forks/composer-dependency-analyser/bin/composer-dependency-analyser.phar"
? Checking Composer compatibility
    > Supported version detected
? No compactor to register
? Mapping paths
  - console.txt > .box.manifests/console.txt
? Adding main file: /shared/backups/forks/composer-dependency-analyser/bin/composer-dependency-analyser
? Adding requirements checker
? Adding binary files
    > 1 file(s)
? Auto-discover files? Yes
? Exclude dev files? Yes
? Adding files
    > 31 file(s)
? Using stub file: /shared/backups/forks/composer-dependency-analyser/stub.php
? Dumping the Composer autoloader
? Removing the Composer dump artefacts
? No compression
? Setting file permissions to 0755
* Done.

No recommendation found.
No warning found.

 // PHAR: 74 files (321.04KB)
 // You can inspect the generated PHAR with the "info" command.

 // Memory usage: 15.80MB (peak: 15.90MB), time: <1sec
```

Reading manifest from PHAR application is no more easy than just invoking your application with `--manifest` option

For example, in our case : 

![box-manifest-1](https://github.com/user-attachments/assets/adbae720-5ee5-49d1-8735-c54f908d63e0)

![box-manifest-2](https://github.com/user-attachments/assets/8c186671-ff4b-453c-89a5-bb80ebfd6435)

**Comment by janedbal** [comment]
- This tool has **zero composer dependencies**, I dont think it needs to be scoped (in phar). 
- Also, we **support analysis of codebases located elsewhere**, so you can install it anywhere and point it to the codebase you want to analyse.


So I dont think this is needed, sorry.

**Comment by llaville** [comment]
Of, as you want, but consider at least to fix #168
